generate/convert.go

```diff
@@ -51,20 +50,26 @@ func (g *generator) convertOperation(
 		return nil, errorf(operation.Position, "%v", err)
 	}
 
-	goTyp, err := g.convertDefinition(
-		name, operation.Name, baseType, operation.Position,
-		operation.SelectionSet, queryOptions, queryOptions)
-
-	if structType, ok := goTyp.(*goStructType); ok {
-		// Override the ordinary description; the GraphQL documentation for
-		// Query/Mutation is unlikely to be of value.
-		// TODO(benkraft): This is a bit awkward/fragile.
-		structType.Description =
-			fmt.Sprintf("%v is returned by %v on success.", name, operation.Name)
-		structType.Incomplete = false
+	// Instead of calling out to convertType/convertDefinition, we do our own
+	// thing, because we want to do a few things differently, and because we
+	// know we have an object type, so we can include only that case.
+	fields, err := g.convertSelectionSet(
+		&prefixList{last: operation.Name}, operation.SelectionSet, baseType, queryOptions)
```

Comment:
It seems like a prefixList constructor could be useful so the caller doesn't need to know anything about the internal structure of the list.

Reply:
ah, sure.